app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorDrawerActivity.kt

```diff
@@ -1,11 +1,104 @@
+import android.view.View
+import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.navigation.findNavController
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.navigateUp
+import androidx.navigation.ui.setupActionBarWithNavController
+import androidx.navigation.ui.setupWithNavController
+import com.example.sharewhatyoucanproject.databinding.ActivityDonordrawerBinding
+import com.example.sharewhatyoucanproject.donor.PostsActivity
+import com.google.android.material.navigation.NavigationView
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.messaging.FirebaseMessaging
 
 class DonorDrawerActivity : AppCompatActivity() {
+    private lateinit var appBarConfiguration: AppBarConfiguration
+    private lateinit var binding: ActivityDonordrawerBinding
+    lateinit var circularProgressIndicator: com.google.android.material.progressindicator.CircularProgressIndicator
+    lateinit var userNames: TextView
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_donordrawer)
+        binding = ActivityDonordrawerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        supportActionBar?.hide()
+        circularProgressIndicator = findViewById(R.id.progress_circular)
+        setSupportActionBar(binding.appBarDrawer.toolbar)
+        val drawerLayout: DrawerLayout = binding.drawerLayout
+        val navView: NavigationView = binding.navView
+        val navController = findNavController(R.id.nav_host_fragment_content_drawer)
+        appBarConfiguration = AppBarConfiguration(
+            setOf(R.id.nav_home),
+            drawerLayout
+        )
+        setupActionBarWithNavController(navController, appBarConfiguration)
+        navView.setupWithNavController(navController)
+        val navigationView1 = findViewById<NavigationView>(R.id.nav_view)
+        val headerview = navigationView1.getHeaderView(0)
+        userNames = headerview.findViewById(R.id.donertv)
+        userNames.text = "" + FirebaseAuth.getInstance().currentUser?.displayName
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel =
+                NotificationChannel("myCh", "Channel1", NotificationManager.IMPORTANCE_HIGH)
+            val manager = getSystemService(NotificationManager::class.java)
+            manager.createNotificationChannel(channel)
+        }
+        FirebaseMessaging.getInstance()
+            .subscribeToTopic("" + FirebaseAuth.getInstance().currentUser?.uid)
+        navigationView1.menu.getItem(1).setOnMenuItemClickListener {
+            drawerLayout.closeDrawers()
+            val i = Intent(this@DonorDrawerActivity, PostsActivity::class.java)
+            startActivity(i)
+            false
+        }
+        navigationView1.menu.getItem(2).setOnMenuItemClickListener {
```

Comment:
you can set a global menu clicked listener and then switch based on the menu item ID. That way it is more readable because the ID will act as a descriptor for which item the user has selected

Reply:
done